### PR TITLE
"evolve_captured" is now using a list instead of a boolean

### LINKED ISSUE #1076: evolve_captured in config.json does not use the specified evolve list
### Expected Behavior
Bot also checks against list of pokemon to evolve

### Actual Behavior
Bot only using `pokemon_to_transfer`

### Steps to Reproduce
Bot captures pokemon and evolves

### Other Information
OS: 
Git Commit: `db5b719dba2faebf9401b929811d0308e2561317`

Excerpt from `config.json`
`"evolve_all": "Pidgey,Weedle,Rattata,Caterpie,Geodude,Ekans,Zubat",`
Log:
```
[22:53:57] Found fort 5420f52faf294cdcbba72166743ef45b.16 at distance 0.02km
[22:53:57] Need to move closer to Pokestop
[                                        ] 0%[22:54:01] Something rustles nearby!
[22:54:01] A Wild Sandshrew appeared! [CP 290] [Potential 0.53]
[22:54:01] IV [Stamina/Attack/Defense] = [15/2/7]
[22:54:05] Using Pokeball (chance: 38.66%)... (0 left!)
[22:54:06] Captured Sandshrew! [CP 290] [15/2/7]
[22:54:06] Sandshrew has been evolved!
```

### PR DESCRIPTION
**Pull request by valerian**
"evolve_captured" is now using a list instead of a boolean, working the same way as "evolve_all"

Fixes #1076 and #1309

### CODE REVIEW

**Comment by douglascamata** [comment]
Did you test both in CLI and JSON?

**Comment by valerian** [comment]
>Did you test both in CLI and JSON?

yes

File: configs/config.json.example [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "use_lucky_egg": false,
     "hatch_eggs": true,
     "longer_eggs_first": true,
-    "evolve_captured": false,
```
Comment (TheSavior) [inline configs/config.json.example]: What behavior should happen for people who haven't updated their config? If false won't work the same as "NONE" then I think we should have a check and if it is a boolean raise an exception to let users know the configuration has changed.

**Comment by valerian** [comment]
Added a parse error for when `evolve_catched` is not a string, or is a boolean-looking string:

> "evolve_captured" should be list of pokemons: use "all" or "none" to match all or none of the pokemons, or use a comma separated list such as "Pidgey,Weedle,Caterpie"

File: configs/config.json.example (hunk 1 above)
Comment (valerian) [inline configs/config.json.example]: `false` will do the same as `"none"`, but `true` will not do the same as `"all"`. Added a parse error to cover that situation, as requested.

**Comment by douglascamata** [comment]
👍

**Comment by DimaVIII** [comment]
https://github.com/PokemonGoF/PokemonGo-Bot/issues/1643

**Comment by bigkraig** [comment]
evolve_captured should be a list in the config example:
```
    "evolve_captured": ["NONE"],
```

**Comment by valerian** [comment]
@bigkraig I just followed the exact same logic as `evolve_all` which is currently like that in the config:

```
    "evolve_all": "NONE",
```

**Comment by valerian** [comment]
@DimaVIII this is not relevant to this pull request, I did only change the conditions for evolving and the config parsing, the rest was untouched.

**Pull request merged**